simulator/src/main/java/com/hedera/block/simulator/BlockStreamSimulatorApp.java

```diff
@@ -71,31 +76,52 @@ public BlockStreamSimulatorApp(
      * @throws IOException if an I/O error occurs
      */
     public void start() throws InterruptedException, BlockSimulatorParsingException, IOException {
-        int delayMSBetweenBlockItems = delayBetweenBlockItems / 1_000_000;
-        int delayNSBetweenBlockItems = delayBetweenBlockItems % 1_000_000;
-
         isRunning.set(true);
         LOGGER.log(System.Logger.Level.INFO, "Block Stream Simulator has started");
 
+        if (streamingMode == StreamingMode.MILLIS_PER_BLOCK) {
+            millisPerBlockStreaming();
+        } else {
+            constantRateStreaming();
+        }
+
+        LOGGER.log(System.Logger.Level.INFO, "Block Stream Simulator has stopped");
+    }
+
+    /**
+     * Returns whether the block stream simulator is running.
+     *
+     * @return true if the block stream simulator is running, false otherwise
+     */
+    public boolean isRunning() {
+        return isRunning.get();
+    }
+
+    /** Stops the block stream simulator. */
+    public void stop() {
+        isRunning.set(false);
+        LOGGER.log(System.Logger.Level.INFO, "Block Stream Simulator has stopped");
+    }
+
+    private void constantRateStreaming()
+            throws InterruptedException, IOException, BlockSimulatorParsingException {
+        int delayMSBetweenBlockItems = delayBetweenBlockItems / 1_000_000;
```

Comment:
Do you think it's worth it to bring out those magic numbers in a variable or in the constants, for readability ? We know what they are and why are needed, but for someone just seeing the codebase could be confusing.